### PR TITLE
Set the 'PLATFORM_OSX' constant during OSX build

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
see #22616

### CODE REVIEW

File: clr.defines.targets [hunk 1]
```diff
@@ -30,5 +30,6 @@
 
         <DefineConstants Condition="'$(TargetsUnix)' == 'true'">$(DefineConstants);PLATFORM_UNIX</DefineConstants>
         <DefineConstants Condition="'$(TargetsWindows)' == 'true'">$(DefineConstants);PLATFORM_WINDOWS</DefineConstants>
+        <DefineConstants Condition="'$(TargetsOSX)' == 'true'">$(DefineConstants);PLATFORM_OSX</DefineConstants>
```
Comment (jkotas) [inline clr.defines.targets]: Can https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/System.Private.CoreLib.csproj#L87 be deleted now?

Comment (AaronRobinsonMSFT) [reply clr.defines.targets]: Sure. Let me do that too.

**Comment by AaronRobinsonMSFT** [comment]
Failures are either known or unrelated to the `PLATFORM_OSX` constant.

**Pull request merged**